### PR TITLE
Add a local perl module to get year last changed

### PR DESCRIPTION
**Pull request by richsalz**
This is used for generating a more-correct copyright statement
for the "build_generated" targets.

Fixes: #13765

### CODE REVIEW

**Comment by richsalz** [comment]
I only started to change one file, `charmap.pl`, and I'm not doing it right.  Ping any Perl folks (@levitte? @mspncp? Bueller?) to give me a hand.

**Comment by mspncp** [comment]
@richsalz I am flattered and cajoled that you count me as a perl guy. Unfortunately, any perl skills which I might have briefly posessed at some time, have almost entirely faded during the last year of not practicing them.

**Comment by richsalz** [comment]
ping @levitte to help me with the import statements needed, now that he's back from holidays and ready for easy things :)

File: crypto/asn1/charmap.pl
Comment (levitte) [inline crypto/asn1/charmap.pl:11]: FindBin finds modules relative to the calling script. Therefore, in this script, this use statement should be:

``` perl
use lib "$FindBin::Bin/../../util/perl";
```

File: util/perl/OpenSSL/copyright.pm
Comment (levitte) [inline util/perl/OpenSSL/copyright.pm:28]: Should be like this:

``` perl
    my $LINE = <$FH>;
    chomp($LINE);
    $YEAR = $LINE if $LINE;
```

... and return $YEAR further down. 

If this isn't done, you'll end up with a year that includes a line ending character.

Comment (richsalz) [reply util/perl/OpenSSL/copyright.pm:28]: wonder why i didn't notice that?  but thanks, added the chomp() stuff.

File: crypto/asn1/charmap.pl
Comment (richsalz) [reply crypto/asn1/charmap.pl:11]: nothing that finds modules relative to a env var, such as top of tree?  sigh, okay.  thanks!

Comment (levitte) [reply crypto/asn1/charmap.pl:11]: Nope.  `FindBin` basically gives you the same as `dirname($0)` in a bash script, you gotta figure out the rest yourself.

BTW, [there's a manual :wink:](https://perldoc.perl.org/FindBin)

Comment (richsalz) [reply crypto/asn1/charmap.pl:11]: Oh man, I just now realized it's a *variable.*  Thanks.

**Comment by richsalz** [comment]
This seems correct now.  @kroeckx did you need fuzz/oids.txt ?  It's not used in openssl, so I removed it, but want to double-check.

**Comment by kroeckx** [comment]
It's used by the fuzzers, don't remove it.

**Comment by richsalz** [comment]
> It's used by the fuzzers, don't remove it.

I thought so, but "git grep" didn't find anything.  I dropped the commit, but if you don't mind telling me where I would appreciate it.

**Comment by kroeckx** [comment]
oss-fuzz uses it, I use it when running it manually.

**Comment by richsalz** [comment]
> oss-fuzz uses it

Yeah, found out the hardway when my CI's failed :)  THANKS!

**Comment by richsalz** [comment]
Was getting a failure in evp_test; rebased and pushed to try again.

**Comment by richsalz** [comment]
ping.

**Comment by richsalz** [comment]
ping.  Does the project want this? @ddvo @mattcaswell

File: fuzz/mkfuzzoids.pl
Comment (mattcaswell) [inline fuzz/mkfuzzoids.pl:14]: Am I missing something? I can't see how this is used at all....can we just delete this line?

Comment (richsalz) [reply fuzz/mkfuzzoids.pl:14]: I was hoping to update the output file to have comments.  But for now, I can put comment-out that line.

Comment (richsalz) [reply fuzz/mkfuzzoids.pl:13]: It turns out that oss-fuzz allows comments, so I updated the script to output the common copyright notice and year.

**Review by mattcaswell — APPROVED** [review]
LGTM

**Comment by mattcaswell** [comment]
Ping @openssl/commmitters. This still needs a second review.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master. Thank you for the contribution.

**Pull request closed without merging**
